int64 instructionID
string message
